deepfence_worker/cronjobs/notification.go

```diff
@@ -30,24 +30,34 @@ func SendNotifications(msg *message.Message) error {
 		log.Error().Msgf("Error getting postgresCtx", err)
 		return nil
 	}
+	wg := new(sync.WaitGroup)
+	wg.Add(len(integrations))
 	for _, integrationRow := range integrations {
-		switch integrationRow.Resource {
-		case utils.ScanTypeDetectedNode[utils.NEO4J_VULNERABILITY_SCAN]:
-			processIntegration[model.Vulnerability](msg, integrationRow)
-		case utils.ScanTypeDetectedNode[utils.NEO4J_SECRET_SCAN]:
-			processIntegration[model.Secret](msg, integrationRow)
-		case utils.ScanTypeDetectedNode[utils.NEO4J_MALWARE_SCAN]:
-			processIntegration[model.Malware](msg, integrationRow)
-		case utils.ScanTypeDetectedNode[utils.NEO4J_COMPLIANCE_SCAN]:
-			processIntegration[model.Compliance](msg, integrationRow)
-			// cloud compliance scans
-			integrationRow.Resource = utils.ScanTypeDetectedNode[utils.NEO4J_CLOUD_COMPLIANCE_SCAN]
-			processIntegration[model.CloudCompliance](msg, integrationRow)
-		}
+		go processIntegrationRow(wg, integrationRow, msg)
```

Comment:
I am not sure we need to do this at all.
What is this solving?

Reply:
I would go one step further and limit to 1, removing `go` routine entirely ;)
Unless we have a good reason, here it feels we are making the code more complex for little benefit, we are actually stressing the DB access & memory usage more than necessary.
Instead we should optimize the sending of HTTP requests (batch + go routines)

Reply:
ok that should minimize frequent neo4j access